src/physics/World.js

```diff
@@ -1001,29 +1001,235 @@ Phaser.Plugin.Isometric.Arcade.prototype = {
 
     },
 
-
     /**
      * Find the distance between a display object (like a Sprite) and the given x/y/z coordinates.
      * The calculation is made from the display objects x/y/z coordinate. This may be the top-left if its anchor hasn't been changed.
      * If you need to calculate from the center of a display object instead use the method distanceBetweenCenters()
      *
      * @method Phaser.Plugin.Isometric.Arcade#distanceToXYZ
-     * @param {any} displayObject - The Display Object to test from.
+     * @param {any} displayObjectBody - The Display Object to test from.
      * @param {number} x - The x coordinate to test to.
      * @param {number} y - The y coordinate to test to.
      * @param {number} z - The y coordinate to test to
      * @return {number} The distance between the object and the x/y coordinates.
      */
-    distanceToXYZ: function (displayObject, x, y, z) {
+    distanceToXYZ: function (displayObjectBody, x, y, z) {
 
-        this._dx = displayObject.x - x;
-        this._dy = displayObject.y - y;
-        this._dz = displayObject.z - z;
+        this._dx = displayObjectBody.x - x;
+        this._dy = displayObjectBody.y - y;
+        this._dz = displayObjectBody.z - z;
```

Comment:
Hopefully specifying displayObjectBody shows where you should pass in sprite.body not sprite. 

This is to deal with a frequent gotcha where you pass an isoSprite and get strange results. This is because it does the calculation from sprite.,sprite.y,sprite.z which are non isometric co-ordinates (x,y) leading to strange results. If you pass in sprite.body, it uses isometric co-ordinates. 

I decided not to explicitly access sprite.isoX so a point3 can still be passed instead of a sprite.body.